### PR TITLE
Karkakol/add active simulcast encodings

### PR DESCRIPTION
**Pull request by karkakol**

### CODE REVIEW

File: MembraneRTC/src/main/java/org/membraneframework/rtc/models/TrackContext.kt [hunk 1]
```diff
@@ -28,6 +35,9 @@ class TrackContext(track: RemoteTrack?, val endpoint: Endpoint, val trackId: Str
     var metadata: Metadata = metadata
         internal set
 
+    var simulcastConfig: SimulcastConfig? = simulcastConfig
```
Comment (graszka22) [inline MembraneRTC/src/main/java/org/membraneframework/rtc/models/TrackContext.kt:38]: also add it to the docs

File: MembraneRTC/src/main/java/org/membraneframework/rtc/models/Endpoint.kt [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
+import org.membraneframework.rtc.events.TracksAdded
 import org.membraneframework.rtc.utils.Metadata
 
 data class Endpoint(
     val id: String,
     val type: String,
     val metadata: Metadata? = mapOf(),
-    val trackIdToMetadata: Map<String, Metadata?> = mapOf()
+    val trackIdToMetadata: Map<String, Metadata?> = mapOf(),
+    val tracks: Map<String, TracksAdded.Data.TrackData>
```
[the remark this answers is not in the document]
Comment (graszka22) [reply MembraneRTC/src/main/java/org/membraneframework/rtc/models/Endpoint.kt:11]: Also an endpoint might be a local endpoint, do we update it in that case?

[the remark this answers is not in the document]
Comment (graszka22) [reply MembraneRTC/src/main/java/org/membraneframework/rtc/models/Endpoint.kt:11]: if it's user facing it doesn't look nice to me (the `TracksAdded.Data.TrackData` type is too long and `TracksAdded` event is not user facing. Consider creating a new type maybe?

File: .idea/inspectionProfiles/Project_Default.xml [hunk 3]
```diff
@@ -7,13 +7,15 @@
     </inspection_tool>
     <inspection_tool class="PreviewApiLevelMustBeValid" enabled="true" level="ERROR" enabled_by_default="true">
       <option name="composableFile" value="true" />
+      <option name="previewFile" value="true" />
```
[the remark this answers is not in the document]
Comment (karkakol) [reply .idea/inspectionProfiles/Project_Default.xml:10]: you are right, .idea shoud be git ignored

File: .idea/gradle.xml [hunk 4]
```diff
@@ -4,17 +4,16 @@
   <component name="GradleSettings">
     <option name="linkedExternalProjectsSettings">
       <GradleProjectSettings>
-        <option name="testRunner" value="GRADLE" />
-        <option name="distributionType" value="DEFAULT_WRAPPED" />
         <option name="externalProjectPath" value="$PROJECT_DIR$" />
-        <option name="gradleJvm" value="Embedded JDK" />
+        <option name="gradleJvm" value="#GRADLE_LOCAL_JAVA_HOME" />
```
[the remark this answers is not in the document]
Comment (karkakol) [reply .idea/gradle.xml:8]: file should be ignored

File: MembraneRTC/src/main/java/org/membraneframework/rtc/events/Event.kt [hunk 5]
```diff
@@ -260,7 +261,13 @@ data class OfferData(val type: ReceivableEventType, val data: Data) : Receivable
 }
 
 data class TracksAdded(val type: ReceivableEventType, val data: Data) : ReceivableEvent() {
-    data class Data(val endpointId: String, val trackIdToMetadata: Map<String, Metadata?>)
+    data class Data(
+        val endpointId: String,
+        val trackIdToMetadata: Map<String, Metadata?>,
+        val tracks: Map<String, TrackData>
```
[the remark this answers is not in the document]
Comment (karkakol) [reply MembraneRTC/src/main/java/org/membraneframework/rtc/events/Event.kt:267]: not necessary, `trackIdToMetadata` is for backward compatibility, i will talk to boys is there any reason to leave it

File: MembraneRTC/src/main/java/org/membraneframework/rtc/models/TrackContext.kt (hunk 1 above)
Comment (karkakol) [reply MembraneRTC/src/main/java/org/membraneframework/rtc/models/TrackContext.kt:38]: good point

File: MembraneRTC/src/main/java/org/membraneframework/rtc/models/Endpoint.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (karkakol) [reply MembraneRTC/src/main/java/org/membraneframework/rtc/models/Endpoint.kt]: unnecessary data is removed

File: MembraneRTC/src/main/java/org/membraneframework/rtc/models/TrackData.kt [hunk 6]
```diff
@@ -0,0 +1,6 @@
+package org.membraneframework.rtc.models
+
+import org.membraneframework.rtc.SimulcastConfig
+import org.membraneframework.rtc.utils.Metadata
+
+data class TrackData(val metadata: Metadata?, val simulcastConfig: SimulcastConfig?)
```
Comment (graszka22) [inline MembraneRTC/src/main/java/org/membraneframework/rtc/models/TrackData.kt:6]: You may document this too if it's user-facing

**Review by graszka22 — COMMENTED** [review]
Also did you check my comment:
> Also an endpoint might be a local endpoint, do we update it in that case?

**Review by graszka22 — APPROVED** [review]

**Pull request merged**